### PR TITLE
[batch] delete pods when they contain no more useful information

### PR DESCRIPTION
**Pull request by danking**
@cseed @jigold I think this resolves our PVC issues modulo batch failure. We still need to add the batch refresh loop.

### CODE REVIEW

File: batch/batch/server/server.py [hunk 1]
```diff
@@ -480,6 +490,7 @@ def is_successful(self):
 
     async def mark_unscheduled(self):
         if self._pod_name:
+            await _delete_pod_by_name(self._pod_name)
```
Comment (jigold) [inline batch/batch/server/server.py]: I think you should move the db update to the function _delete_pod_by_name

File: batch/batch/server/server.py [hunk 2]
```diff
@@ -255,18 +255,27 @@ def _has_next_task(self):
...
+                v1.delete_namespaced_pod,
+                name,
+                HAIL_POD_NAMESPACE,
+                _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS)
+        except kube.client.rest.ApiException as err:
+            if err.status == 404:
+                pass
+            else:
+                log.exception(f'pod deletion ({name}) threw {err}, but we will not fail')
+                return err
```
Comment (jigold) [inline batch/batch/server/server.py]: I don't think you need to return anything here. Not used at all.

File: batch/batch/server/server.py [hunk 3]
```diff
@@ -299,6 +308,7 @@ def _has_next_task(self):
         self._task_idx += 1
         self._current_task = self._tasks[self._task_idx] if self._task_idx < len(self._tasks) else None
 
+        await _delete_pod_by_name(self._pod_name)
```
Comment (jigold) [inline batch/batch/server/server.py]: The line below should prob just be `self._pod_name = None` without the if statement.

File: batch/batch/server/server.py [hunk 4]
```diff
@@ -480,6 +490,7 @@ def is_successful(self):
 
     async def mark_unscheduled(self):
         if self._pod_name:
+            await _delete_pod_by_name(self._pod_name)
             await db.jobs.update_record(self.id, pod_name=None)
             self._pod_name = None
```
Comment (jigold) [inline batch/batch/server/server.py]: I feel like this assignment should be part of the function delete_pod_by_name.

File: batch/batch/server/server.py (hunk 3 above)
Comment (jigold) [reply batch/batch/server/server.py]: Should this be after we update the database below (update_with_log_ec)?

File: batch/batch/server/server.py (hunk 1 above)
Comment (danking) [reply batch/batch/server/server.py]: I'd have to sometimes do update_record and sometimes update_with_log_ec, I think it's better to keep them separate for now.

**Pull request closed without merging**